be/src/exec/tablet_sink.cpp

```diff
@@ -186,80 +267,89 @@ void NodeChannel::cancel() {
 
     closure->ref();
     closure->cntl.set_timeout_ms(_rpc_timeout_ms);
-    _stub->tablet_writer_cancel(&closure->cntl,
-                                &request,
-                                &closure->result,
-                                closure);
+    _stub->tablet_writer_cancel(&closure->cntl, &request, &closure->result, closure);
     request.release_id();
 
-    // reset batch
-    _batch.reset();
+    // Beware of the destruct sequence. RowBatches will use mem_trackers(include ancestors).
+    // Delete RowBatches here is a better choice to reduce the potential of dtor errors.
+    {
+        std::lock_guard<std::mutex> lg(_pending_batches_lock);
+        std::queue<AddBatchReq> empty;
+        std::swap(_pending_batches, empty);
+        _cur_batch.reset();
+    }
 }
 
-Status NodeChannel::_wait_in_flight_packet() {
-    if (!_has_in_flight_packet) {
-        return Status::OK();
+int NodeChannel::try_send_and_fetch_status() {
```

Comment:
Why not using bool as return value?

Reply:
I want the consumer to have global observation,  sum() is the simplest way to do it.
And maybe we can add more status(>0) to let consumer know more infos about channels. 
But I think the preliminary design should be simple.
So current design of status value is "0-channel stopped; 1-still working".